### ISSUE TITLE
Add support for "Member of" triggers

### ISSUE THREAD

**Issue by Dbzman** [seaside1/jrule#22]
As a follow-up from the [JRule thread](https://community.openhab.org/t/jrule-openhab-rules-using-java/121441/61), support for ["Member of" triggers](https://www.openhab.org/docs/configuration/rules-dsl.html#member-of-triggers) would be super handy. Also the JRule docs don't explicitly make it obvious that groups are handled in a way that only events are triggered when the group's value changes (not from it's members). But the docs is a rather easy fix, I guess.

**Comment by seaside1** [comment]
I guess you have seen this? 
https://github.com/seaside1/jrule/pull/31

Member of <group> received command [<command>]
Member of <group> received update [<state>]
Member of <group> changed [from <state>] [to <state>]

If I would implement this what would you like to have? Like for group items, have a trigger for member of like this:
````java
  @JRuleName("testmemberOf")
    @JRuleWhen(item = _gMyGroup.ITEM, trigger = _gMyGroup.TRIGGER_MEMBER_OF_RECEIVED_COMMAND)
    @JRuleWhen(item = _gMyGroup.ITEM, trigger = _gMyGroup.TRIGGER_MEMBER_OF_RECEIVED_UPDATE)
    @JRuleWhen(item = _gMyGroup.ITEM, trigger = _gMyGroup.TRIGGER_CHANGED)
````

I guess it would add complexity to allow for triggers like TRIGGER_CHANGED_FROM_ON_TO_OFF etc.

**Comment by Dbzman** [comment]
Yes, I've seen this but I didn't think that far that it would pave the way for the feature here. :)

I think you're right. Having additional triggers would add complexity. What about that?
```
@JRuleName("testmemberOf")
@JRuleWhen(membersOf = _gMyGroup.ITEM, trigger = _gMyGroup.TRIGGER_RECEIVED_COMMAND)
@JRuleWhen(membersOf = _gMyGroup.ITEM, trigger = _gMyGroup.TRIGGER_RECEIVED_UPDATE)
@JRuleWhen(membersOf = _gMyGroup.ITEM, trigger = _gMyGroup.TRIGGER_CHANGED)
```
That way we can re-use the triggers but make it explicit that it acts on `membersOf` the given item. (Or maybe `membersOfGroup` is even more explicit)

**Comment by seaside1** [comment]
Looks good, would still need to have the item = since it is used to filter out events on the eventbus (hashmap only containing the items that we should subscribe on). Would maybe be possible to use the memberOf in that subscription as well. I can try and take a look

**Comment by Dbzman** [comment]
I see. It would also be fine for me if the member of thingy was a a separate flag or so like:
```
@JRuleWhen(item = _gMyGroup.ITEM, trigger = _gMyGroup.TRIGGER_RECEIVED_COMMAND, memberOf=true)
```
I could also imagine to have a kind of `triggerFor` option which is maybe set to `ITEM` as default and it's possible to set it to `MEMBER_OF`, like this:
```
@JRuleWhen(item = _gMyGroup.ITEM, trigger = _gMyGroup.TRIGGER_RECEIVED_COMMAND, triggerFor=MEMBER_OF)
```
Oh, and I realized that I used the plural form (`membersOf`) in my previous post, which maybe does not make too much sense.

**Comment by querdenker2k** [comment]
I think JRuleWhen is very overloaded. What about splitting this annotation into something like JRuleWhenItemChanged, JRuleWhenItemUpdated, JRuleWhenMemberOfChanged, ...

**Comment by querdenker2k** [comment]
I would now suggest (as well) something like
```
@JRuleWhenItemChange(item = _MySwitchGroup.ITEM, memberOf=true)
@JRuleWhenItemReceivedCommand(item = _MySwitchGroup.ITEM, memberOf=true)
@JRuleWhenItemReceivedUpdate(item = _MySwitchGroup.ITEM, memberOf=true)
```

**Comment by seaside1** [comment]
Closing this as it is fixed in pr https://github.com/seaside1/jrule/pull/66

**Comment by Dbzman** [comment]
Thanks a lot for implementing this feature! I'm looking forward to try this out. :)

**Comment by seaside1** [comment]
You should probably wait a bit. There are  a number of major prs that will fix a lot of things (thinking of https://github.com/seaside1/jrule/pull/77 for instance)
